include/tvm/relay/dataflow_matcher.h

```diff
@@ -97,15 +97,29 @@ Expr RewritePatterns(Array<DFPatternCallback> callbacks, Expr expr, IRModule mod
  * \brief Partition all matches of a DFPattern inside an Expr into separate Function calls
  *
  * \param pattern The pattern to match
- * \param expr The expression to patition
+ * \param expr The expression to partition
  * \param attrs A set of parameter names and values to apply to the partitioned function
  * \param check A callback function for checking more complicated properties of the matched
  * expressions, returns true if the match is accepted and false otherwise
  *
- * \return Return the paritioned Expr.
+ * \return Return the partitioned Expr.
  */
 Expr PartitionPattern(DFPattern pattern, Expr expr, Map<String, ObjectRef> attrs, PackedFunc check);
 
+/*!
+ * \brief Partition all matches of a DFPattern inside an Expr into separate Function calls
```

Comment:
You'll need to explain the 'hierarchical order' part here, perhaps explain they are expected to be in most-specific to most-general form and the first pattern to succeed is taken.

Reply:
Oh now that i look at the impl i see it's not that at all. So yeah will need to explain :-)